app/src/main/java/com/kickstarter/libs/UserCurrency.kt

```diff
@@ -0,0 +1,132 @@
+     *
+     * @param initialValue Value to display, local to the project's currency.
+     * @param project The project to use to look up currency information.
+     */
+    fun format(initialValue: Float, project: Project, symbol: String): String {
+        return format(initialValue, project, false, RoundingMode.DOWN, symbol)
+    }
+
+    /**
+     * Returns a currency string appropriate to the user's locale and preferred currency.
+     *
+     * @param initialValue Value to display, local to the project's currency.
+     * @param project The project to use to look up currency information.
+     * @param excludeCurrencyCode If true, hide the currency code, even if that makes the returned value ambiguous.
+     * This is used when space is constrained and the currency code can be determined elsewhere.
+     */
+    fun format(initialValue: Float, project: Project,
+               excludeCurrencyCode: Boolean, symbol: String): String {
+
+        return format(initialValue, project, excludeCurrencyCode, RoundingMode.DOWN, symbol)
+    }
+
+    /**
+     * Returns a currency string appropriate to the user's locale and preferred currency.
+     *
+     * @param initialValue Value to display, local to the project's currency.
+     * @param project The project to use to look up currency information.
+     * @param excludeCurrencyCode If true, hide the currency code, even if that makes the returned value ambiguous.
+     * This is used when space is constrained and the currency code can be determined elsewhere.
+     */
+    fun format(initialValue: Float, project: Project,
+               excludeCurrencyCode: Boolean, roundingMode: RoundingMode, symbol: String): String {
+
+        val currencyOptions = userCurrencyOptions(initialValue, project, symbol)
+
+        val showCurrencyCode = showCurrencyCode(currencyOptions, excludeCurrencyCode)
+
+        val numberOptions = NumberOptions.builder()
+                .currencyCode(if (showCurrencyCode) currencyOptions.currencyCode() else "")
+                .currencySymbol(currencyOptions.currencySymbol())
+                .roundingMode(roundingMode)
+                .build()
+
+        return NumberUtils.format(currencyOptions.value(), numberOptions)
+    }
+
+    /** Show's the project in the user's preferred currency. If the user has no preferred currency the project is shown
+     * in $ as a default if the user is in the US. If the user is loacated outside of the US the default will show as
+     * $US.
+     */
+    private fun userCurrencyOptions(value: Float, project: Project, symbol: String): KSCurrency.CurrencyOptions {
+        val fxRate = project.fx_rate()
+        val config = this.currentConfigType.getConfig()
+
+        return if (config.countryCode() == "XX") {
+            KSCurrency.CurrencyOptions.builder()
+                    .country(project.country())
+                    .currencyCode("")
```

Comment:
What is `currencyCode` used for and why do you pass "" for both the if and the else

Reply:
So after some deep diving, we actually we were actually never using it because it would've shown in the projects currency. Now we actually capture it and map it to the showCurrencySymbol method where we map to the display strings we want to show the user. I just removed it. Great catch!